flake.nix

```diff
@@ -0,0 +1,167 @@
+                      self.callCabal2nix "ghc-typelits-natnormalise"
+                      ghc-typelits-natnormalise { };
+                    # due to random, hashable on GHC-9.2.1
+                    "linear" = haskellLib.doJailbreak super.linear_1_21_7;
+                    # loosen base bound on GHC-9.2.1
+                    "some" = haskellLib.doJailbreak super.some;
+                    # loosen base bound on GHC-9.2.1
+                    "universe-base" = super.universe-base_1_1_3;
+                    # loosen base bound on GHC-9.2.1
+                    "vec" = haskellLib.doJailbreak super.vec;
+                  }));
+          });
+        };
+
+        parseCabalProject = import ./parse-cabal-project.nix;
+        categorifierPackages = parseCabalProject ./cabal.project;
+        categorifierPackageNames =
+          builtins.map ({ name, ... }: name) categorifierPackages;
+        haskellOverlay = self: super:
+          builtins.listToAttrs (builtins.map ({ name, path }: {
+            inherit name;
+            value = self.callCabal2nix name (./. + "/${path}") { };
+          }) categorifierPackages);
+
+        # see these issues and discussions:
+        # - https://github.com/NixOS/nixpkgs/issues/16394
+        # - https://github.com/NixOS/nixpkgs/issues/25887
+        # - https://github.com/NixOS/nixpkgs/issues/26561
+        # - https://discourse.nixos.org/t/nix-haskell-development-2020/6170
+        fullOverlays = [
+          overlay_deps
+          (final: prev: {
+            haskellPackages = prev.haskellPackages.override (old: {
+              overrides =
+                final.lib.composeExtensions (old.overrides or (_: _: { }))
+                haskellOverlay;
+            });
+          })
+        ];
+
+      in {
+        # This package set is only useful for CI build test.
+        # In practice, users will create a development environment composed by overlays.
+        packages = let
+          packagesOnGHC = ghcVer:
+            let
+              overlayGHC = final: prev: {
+                haskellPackages = prev.haskell.packages.${ghcVer};
+              };
+
+              newPkgs = import nixpkgs {
+                overlays = [ overlayGHC (concat.overlay.${system}) ]
+                  ++ fullOverlays;
+                inherit system;
+                config.allowBroken = true;
+              };
+
+              individualPackages = builtins.listToAttrs (builtins.map
+                ({ name, ... }: {
+                  name = ghcVer + "_" + name;
+                  value = builtins.getAttr name newPkgs.haskellPackages;
+                }) categorifierPackages);
+
+              allEnv = let
+                # CI error: https://github.com/con-kitty/categorifier/runs/6457812761?check_suite_focus=true#step:5:8545
+                excluded = [
+                  "categorifier-concat-extensions-integration-test"
+                  "categorifier-vec-integration-test"
+                ];
```

Comment:
You should be able to remove these exclusions if you rebase on the latest master ... I hope! These definitely block merge.

Reply:
removed and the build passed!